pkg/conn/basedb.go

```diff
@@ -66,27 +66,23 @@ func (d *defaultDBProvider) Apply(config config.DBConfig) (*BaseDB, error) {
 
 // BaseDB wraps *sql.DB
 type BaseDB struct {
+	sync.Mutex
 	DB *sql.DB
 
-	lock *sync.Mutex
-
 	// hold all db connections generated from this BaseDB
-	conns []*BaseConn
+	conns map[*BaseConn]bool
```

Comment:
if you do not need the _value_ of the `map` (just like `set` in other languages), you can just use `map[*BaseConn]struct{}`.

Reply:
ok, it looks better